setup.cfg

```diff
@@ -32,21 +32,25 @@ package_dir=
     = src
 packages = find:
 python_requires = >=3.7
-setup_requires = setuptools >= 45.2.0
 zip_safe = False
 
 [options.packages.find]
 where = src
 
 [options.extras_require]
 test =
```

Comment:
Logically it makes sense to separate these out but is is necessarily worth it? The additional install time is so trivial for the small amount of dependencies so I'm just not sure its really worth the split. It also makes the CI caching harder.

Reply:
What are you talking about?

Reply:
Oh, you mean the separation of test and typing dependencies? That wasn't actually done by me, and I'm still not 100% convinced that we shouldn't be using pre-commit for that. It's not critical either way, and I don't see any particular issues that it would cause for caching. I could of course be convinced otherwise.